docs/egress.md

```diff
@@ -41,6 +41,10 @@ You may be interested in using this capability if any of the following apply:
 - You want to force outgoing external connections to leave the cluster via
   certain Nodes, for security controls, or due to network topology restrictions.
 
+**Note**: A dummy interface `antrea-egress0` is automatically created on the
+Node hosting the egress IP, the interface is intended to be down and egress
+traffic will not flow through it but the interface determined by the route table.
```

Comment:
Considering the context, this sentence seems out of place and disrupts "What is Egress?". Could we relocate it to the end of https://github.com/antrea-io/antrea/blob/main/docs/egress.md#egressip?

Reply:
Given the following context, we can keep it at same location !
https://github.com/antrea-io/antrea/blob/main/docs/egress.md#egressip:~:text=Starting%20with%20Antrea%20v1.2%2C%20high,IP%20will%20be%20selected%20randomly.

Reply:
I'm not if you agree with the comment or not. Your reply seems to agree but you ping me for another review without making any change.

Reply:
> Given the following context, we can keep it at same location ! https://github.com/antrea-io/antrea/blob/main/docs/egress.md#egressip:~:text=Starting%20with%20Antrea%20v1.2%2C%20high,IP%20will%20be%20selected%20randomly.

I meant, can we keep at the present location not at the end of https://github.com/antrea-io/antrea/blob/main/docs/egress.md#egressip, given that the section near the end explain about HA ?

Reply:
The "egressip" section is not talking about HA but everything about egress IPs including a bit technical details:
> Starting with Antrea v1.2, high availability is provided automatically when the egressIP is allocated from an externalIPPool, i.e. when the externalIPPool is specified. If the Node hosting the egressIP fails, another Node will be elected (from among the remaining Nodes selected by the nodeSelector of the externalIPPool) as the new egress Node of this Egress. It will take over the IP and send layer 2 advertisement (for example, Gratuitous ARP for IPv4) to notify the other hosts and routers on the network that the MAC address associated with the IP has changed.

And the dummy interface is indeed only used to host the IPs in HA mode. I think it's more natural to talk about the dummy interface following the above explanation:

> Starting with Antrea v1.2, high availability is provided automatically when the egressIP is allocated from an externalIPPool, i.e. when the externalIPPool is specified. If the Node hosting the egressIP fails, another Node will be elected (from among the remaining Nodes selected by the nodeSelector of the externalIPPool) as the new egress Node of this Egress. It will take over the IP and send layer 2 advertisement (for example, Gratuitous ARP for IPv4) to notify the other hosts and routers on the network that the MAC address associated with the IP has changed. **A dummy interface `antrea-egress0` is automatically created on the Node hosting the egress IP, the interface is intended to be down and egress traffic will not flow through it but the interface determined by the route table.**

By contrast, doesn't the addition of the implementation detail look weird when placed in-between **the purpose of Egress** below?

> You may be interested in using this capability if any of the following apply:
>
> * A consistent IP address is desired when specific Pods connect to services outside of the cluster, for source tracing in audit logs, or for filtering by source IP in external firewall, etc.
>
> * You want to force outgoing external connections to leave the cluster via certain Nodes, for security controls, or due to network topology restrictions.
>
> **Note: A dummy interface antrea-egress0 is automatically created on the Node hosting the egress IP, the interface is intended to be down and egress traffic will not flow through it but the interface determined by the route table.**
>
> This guide demonstrates how to configure Egress to achieve the above result.

Reply:
I see your point, I was also thinking about static egress, however In the static egress case, the antrea-egress0 dummy interface exists but is not used and It is utilised when ExternalIPPools are involved, primarily to provide high availability.